pkg/controller/installation/core_controller.go

```diff
@@ -1626,6 +1627,20 @@ func getOrCreateTyphaNodeTLSConfig(cli client.Client, certificateManager certifi
 func (r *ReconcileInstallation) setDefaultsOnFelixConfiguration(ctx context.Context, install *operator.Installation, fc *crdv1.FelixConfiguration, reqLogger logr.Logger) (bool, error) {
 	updated := false
 
+	if install.Spec.CalicoNetwork.LinuxDataplane != nil {
```

Comment:
Hmmm, this is in a function called `setDefaults...`, so since this always sets the config it seems like we should change the function name or create a new function. I think it'd be good to include a comment about why we're ok setting this compared to how we handle other FelixConfig options where we'll only set defaults.